src/__tests__/evaluator.test.ts

```diff
@@ -77,4 +85,157 @@ describe('Evaluator', () => {
+        variationToTargetMap: [
+          { variation: 'true', targets: [], targetSegments: ['and-segment'] },
+        ],
+        variations: [
+          { identifier: 'true', name: 'True', value: 'true' },
+          { identifier: 'false', name: 'False', value: 'false' },
+        ],
+        version: 1,
+      };
+
+      await repo.setFlag('boolflag_or', feature_config_or);
+      await repo.setFlag('boolflag_and', feature_config_and);
+    }
+    async function loadSegments(repo: StorageRepository) {
+      const segment_or: Segment = {
+        identifier: 'or-segment',
+        name: 'is_harness_or_somethingelse_email_OR',
+        environment: 'Production',
+        // only 1 servingRules needs to be true (OR)
+        servingRules: [
+          {
+            ruleId: 'this_or_rule_with_lower_priority_should_be_ignored',
+            priority: 7,
+            clauses: [
+              {
+                attribute: 'email',
+                op: 'ends_with',
+                values: ['@harness.io'],
+                negate: false,
+              },
+            ],
+          },
+          {
+            ruleId: 'rule1',
+            priority: 1,
+            clauses: [
+              {
+                attribute: 'email',
+                op: 'ends_with',
+                values: ['@harness.io'],
+                negate: false,
+              },
+            ],
+          },
+          {
+            ruleId: 'rule2',
+            priority: 2,
+            clauses: [
+              {
+                attribute: 'email',
+                op: 'ends_with',
+                values: ['@somethingelse.com'],
+                negate: false,
+              },
+            ],
+          },
+        ],
+        version: 2,
+      };
+      const segment_and: Segment = {
+        identifier: 'and-segment',
+        name: 'is_a_harness_developer_test_AND',
+        environment: 'Production',
+        servingRules: [
+          {
+            ruleId: 'rule1',
+            priority: 1,
+            // all clauses need to be true (AND)
+            clauses: [
+              {
+                attribute: 'email',
+                op: 'ends_with',
+                values: ['@harness.io'],
+                negate: false,
+              },
+              {
+                attribute: 'role',
+                op: 'equal',
+                values: ['developer'],
+                negate: false,
+              },
+            ],
+          },
+        ],
+      };
+      await repo.setSegment('or-segment', segment_or);
+      await repo.setSegment('and-segment', segment_and);
+    }
+
+    const cache = new SimpleCache();
+    const repository = new StorageRepository(cache);
+    const evaluator = new Evaluator(repository, logger);
+
+    await loadFlags(repository);
+    await loadSegments(repository);
+
+    const target = {
+      identifier: name,
+      name: name,
+      attributes: {
+        email: email,
+        role: role,
```

Comment:
You can short hand these
```suggestion
      name,
      attributes: {
        email,
        role,
```